shared/chat/blocking/block-modal.tsx

```diff
@@ -0,0 +1,182 @@
+import * as React from 'react'
+import * as Kb from '../../common-adapters'
+import * as Styles from '../../styles'
+
+const todo = () => console.log('TODO')
+type Props = {
+  team?: string
+  adder: string
+  others?: Array<string>
+}
+
+type CheckboxRowProps = {
+  text: React.ReactNode
+  onCheck: (boolean) => void
+  more?: React.ReactNode
+  info?: string
+  checked: boolean
+}
+const CheckboxRow = (props: CheckboxRowProps) => {
+  const [infoShowing, setInfoShowing] = React.useState(false)
+  return (
+    <>
+      <Kb.Box2 direction="horizontal" alignItems="center" fullWidth={true} style={styles.checkBoxRow}>
+        <Kb.Switch
+          color="red"
+          onClick={() => props.onCheck(!props.checked)}
+          on={props.checked}
+          label={props.text}
+          labelSubtitle={infoShowing ? props.info : undefined}
+          gapSize={Styles.globalMargins.tiny}
+          style={styles.shrink}
+        />
+        <Kb.Box style={styles.iconBox} />
+        {props.info && !infoShowing && (
+          <Kb.Icon type="iconfont-question-mark" color="grey" onClick={() => setInfoShowing(true)} />
+        )}
+      </Kb.Box2>
+    </>
+  )
+}
+const reasons = ["I don't know this person", 'Spam', 'Harassment', 'Obscene material', 'Other...']
+const BlockModal = (props: Props) => {
+  const onFinish = todo
+  const [blockTeam, setBlockTeam] = React.useState(true)
+  const [blockAdder, setBlockAdder] = React.useState(true)
+  const [hideAdder, setHideAdder] = React.useState(true)
+  const [report, setReport] = React.useState(false)
+  const [includeTranscript, setIncludeTranscript] = React.useState(false)
+  const [reportReason, setReportReason] = React.useState(reasons[0])
+  const [extraNotes, setExtraNotes] = React.useState('')
+  const [otherBlocked, setOtherBlocked] = React.useState(new Map<string, boolean>())
+
+  const setBlockedOther = (username: string, block: string) => (blocked: boolean) => {
+    otherBlocked.set(`${username}:${block}`, blocked)
+    setOtherBlocked(new Map(otherBlocked))
```

Comment:
Since you are making a new map anyway, this could also be:

```
setOtherBlocked([...otherBlocked, [`${username}:${block}`, blocked]])
```

But why not just put all of these in the redux store? They are gonna need to go through the store later anyway.

Reply:
It'll go in the redux store when the user presses "Finish" - it feels dirty to me to be putting it there before they "save" their decisions.